@here/harp-mapview/lib/MapView.ts

```diff
@@ -2968,6 +2959,17 @@ export class MapView extends THREE.EventDispatcher {
 
         this.m_screenCamera.position.z = 1;
         this.m_screenCamera.near = 0;
+
+        // When calculating clip planes account for the highest building on the earth,
+        // multipling its height by projection scalling factor. This approach assumes
+        // constantHeight property of extruded polygon technique is set as default false,
+        // otherwise the near plane margins will be bigger then required, but still correct.
+        const projectionScale = this.projection.getScaleFactor(this.camera.position);
+        const maxHeight = EarthConstants.MAX_BUILDING_HEIGHT * projectionScale;
+        // Copy all properties from new view ranges to our readonly object.
+        // This allows to keep all view ranges references valid and keeps up-to-date
+        // information within them. Works the same as copping all properties one-by-one.
+        this.m_visibleTiles.updateClipPlanes(maxHeight);
```

Comment:
I see no point of updating clip planes here. The evaluation using EarthConstants.MAX_BUILDING_HEIGHT as default buildings/features height is done for broad phase tiles elimination (frustum culling) and thus allows to see tiles on globe with high geometry behind earth horizon (surface), without maxHeight (even naive) defined is may clip those tiles out in the tile rejection phase, so those geometries (buildings) may be never visible or actually pop-out immediately from behind horizon.

Reply:
Agreed, removed.